internal/commands/pull/command.go

```diff
@@ -138,6 +165,18 @@ func (cmd *Command) Handler(profile *user.Profile, ui terminal.UI, clients cli.C
 		ui.Print(terminal.NewDebugLog("Fetched hosting assets"))
 	}
 
+	successfulTemplateWrites := make([]string, 0, len(clientZipPkgs))
+	for templateID, templateZipPkg := range clientZipPkgs {
```

Comment:
you should add a `TODO(REALMC-9452)` here about `iterating over clientZipPkgs in a deterministic order`